### PR TITLE
create blocs, models, services, custom card

### PR DESCRIPTION
**Pull request by tranminhthuan962**

### CODE REVIEW

File: lib/src/app.dart [hunk 1]
```diff
@@ -1,25 +1,41 @@
 import 'package:flutter/material.dart';
-
+import 'package:flutter_bloc/flutter_bloc.dart';
+import 'package:http/http.dart' as http;
+import '/../src/services/detail_service/app_detail_service.dart';
```
Comment (thanhphamtrung) [inline lib/src/app.dart]: insert a line break

File: lib/src/app.dart [hunk 2]
```diff
@@ -1,25 +1,41 @@
...
-              body: Center(child: Text(StringConstants.notFound))),
-        );
-      },
+    final httpClient = http.Client();
+    return MultiBlocProvider(
+      providers: [
+        BlocProvider(
+            create: (context) =>
+                DetailBloc(service: AppCoinDetailService(httpClient))
+                  ..add(DetailRequested())),
```
Comment (thanhphamtrung) [inline lib/src/app.dart]: you should add this event on screen

File: lib/src/blocs/detail_screen_bloc/detail_bloc.dart [hunk 3]
```diff
@@ -0,0 +1,19 @@
+import 'package:flutter_bloc/flutter_bloc.dart';
```
Comment (thanhphamtrung) [inline lib/src/blocs/detail_screen_bloc/detail_bloc.dart:1]: line break

File: lib/src/routes/route_genertor.dart [hunk 4]
```diff
@@ -1,8 +1,8 @@
 import 'package:flutter/material.dart';
+import '/../src/screens/home_screen/home_screen.dart';
```
Comment (thanhphamtrung) [inline lib/src/routes/route_genertor.dart]: line break

File: lib/src/screens/detail_screen/detail_screen.dart [hunk 5]
```diff
@@ -1,5 +1,10 @@
 import 'package:flutter/cupertino.dart';
 import 'package:flutter/material.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+import 'package:intl/intl.dart' as intl;
+import '/../src/blocs/detail_screen_bloc/detail_bloc.dart';
```
Comment (thanhphamtrung) [inline lib/src/screens/detail_screen/detail_screen.dart]: .

File: lib/src/services/detail_service/app_detail_service.dart [hunk 6]
```diff
@@ -0,0 +1,30 @@
+import 'dart:convert';
+import 'dart:async';
+import 'package:http/http.dart' as http;
+import '/../src/config/app_config.dart';
```
Comment (thanhphamtrung) [inline lib/src/services/detail_service/app_detail_service.dart]: .

File: lib/src/services/detail_service/detail_service.dart [hunk 7]
```diff
@@ -0,0 +1,10 @@
+import 'package:http/http.dart' as http;
+import '/../src/models/detail_model.dart';
```
Comment (thanhphamtrung) [inline lib/src/services/detail_service/detail_service.dart]: .

File: lib/src/widgets/custom_card_detail.dart [hunk 8]
```diff
@@ -0,0 +1,244 @@
+import 'package:flutter/cupertino.dart';
+import 'package:flutter/foundation.dart';
+import 'package:flutter/material.dart';
+import '/../src/constants/color_constants.dart';
```
Comment (thanhphamtrung) [inline lib/src/widgets/custom_card_detail.dart]: .

File: lib/src/widgets/custom_card_detail.dart [hunk 9]
```diff
@@ -0,0 +1,244 @@
...
+                  '($symbol)'.toUpperCase(),
+                  style: TextStyle(
+                      fontSize: coinFontSize, fontWeight: screenTextFontWeight),
+                ),
+              ],
+            ),
+            Padding(
+              padding: EdgeInsets.only(
+                top: screenPadding,
+                // left: screenPadding + 30.0,
```
Comment (thanhphamtrung) [inline lib/src/widgets/custom_card_detail.dart]: remove

File: lib/src/widgets/custom_card_detail.dart [hunk 10]
```diff
@@ -0,0 +1,244 @@
...
+            const CustomLine(),
+            Expanded(
+              child: Column(
+                mainAxisAlignment: MainAxisAlignment.start,
+                crossAxisAlignment: CrossAxisAlignment.start,
+                children: [
+                  Text(
+                    StringConstants.textDescription,
+                    style: TextStyle(
+                        fontSize: 22.0, fontWeight: screenTextFontWeight),
```
Comment (thanhphamtrung) [inline lib/src/widgets/custom_card_detail.dart:205]: why didn't you use Theme?

File: lib/src/widgets/custom_card_detail.dart [hunk 11]
```diff
@@ -0,0 +1,244 @@
...
+
+class CustomLine extends StatelessWidget {
+  const CustomLine({
+    Key? key,
+  }) : super(key: key);
+
+  @override
+  Widget build(BuildContext context) {
+    return Container(
+      margin: const EdgeInsets.symmetric(vertical: 10.0),
```
Comment (thanhphamtrung) [inline lib/src/widgets/custom_card_detail.dart]: layout variable

**Review by thanhphamtrung — COMMENTED** [review]
line break
layout variable
not remove unuse code
=> 30k

File: lib/src/app.dart (hunk 1 above)
Comment (tranminhthuan962) [reply lib/src/app.dart]: Does your mind insert a line break between "package in flutter" and  "package project"?

File: test/src/screens/detail_screen/unit_tests/screens/detail_screen_test.dart [hunk 12]
```diff
@@ -0,0 +1,47 @@
...
+  //     home: const DetailScreen(),
+  //     navigatorObservers: [mockNa],
+  //   ));
+  //
+  //   final iconButton = find.byType(IconButton);
+  //   expect(iconButton, findsOneWidget);
+  //   await buttonTest.press(iconButton);
+  //   verify(mockNa.navigator);
+  // });
+}
```
Comment (tranminhthuan962) [inline test/src/screens/detail_screen/unit_tests/screens/detail_screen_test.dart]: Sorry, I haven't done this file, I made a mistake  in add this file

File: lib/src/widgets/custom_card_detail.dart (hunk 10 above)
Comment (tranminhthuan962) [reply lib/src/widgets/custom_card_detail.dart:205]: I'm not familiar with using Theme

File: lib/src/app.dart (hunk 2 above)
Comment (tranminhthuan962) [reply lib/src/app.dart]: I've done

File: lib/src/blocs/detail_screen_bloc/detail_bloc.dart (hunk 3 above)
Comment (tranminhthuan962) [reply lib/src/blocs/detail_screen_bloc/detail_bloc.dart:1]: I've done

File: lib/src/routes/route_genertor.dart (hunk 4 above)
Comment (tranminhthuan962) [reply lib/src/routes/route_genertor.dart]: I've done

File: lib/src/screens/detail_screen/detail_screen.dart (hunk 5 above)
Comment (tranminhthuan962) [reply lib/src/screens/detail_screen/detail_screen.dart]: I've done

File: lib/src/services/detail_service/detail_service.dart (hunk 7 above)
Comment (tranminhthuan962) [reply lib/src/services/detail_service/detail_service.dart]: I've done

File: lib/src/widgets/custom_card_detail.dart (hunk 8 above)
Comment (tranminhthuan962) [reply lib/src/widgets/custom_card_detail.dart]: I've done

File: lib/src/widgets/custom_card_detail.dart (hunk 9 above)
Comment (tranminhthuan962) [reply lib/src/widgets/custom_card_detail.dart]: I've done

File: lib/src/widgets/custom_card_detail.dart (hunk 11 above)
Comment (tranminhthuan962) [reply lib/src/widgets/custom_card_detail.dart]: I've done

**Review by thanhphamtrung — DISMISSED** [review]
LGTM

File: lib/main.dart [hunk 13]
```diff
@@ -1,6 +1,7 @@
 import 'package:flutter/material.dart';
 
 import '/../src/app.dart';
+
```
Comment (thanhphamtrung) [inline lib/main.dart:4]: ?

File: lib/src/app.dart [hunk 14]
```diff
@@ -9,6 +9,10 @@ import '/../src/blocs/search_bloc/search_bloc.dart';
 import '/../src/constants/name_routes_constants.dart';
 import '/../src/routes/route_genertor.dart';
 
+import '/../src/blocs/detail_screen_bloc/detail_bloc.dart';
+
```
Comment (thanhphamtrung) [inline lib/src/app.dart]: ?

File: lib/src/blocs/detail_screen_bloc/detail_bloc.dart [hunk 15]
```diff
@@ -0,0 +1,21 @@
...
+
+import '/../src/blocs/detail_screen_bloc/detail_event.dart';
+import '/../src/blocs/detail_screen_bloc/detail_state.dart';
+
+import '/../src/services/detail_service/detail_service.dart';
```
Comment (thanhphamtrung) [inline lib/src/blocs/detail_screen_bloc/detail_bloc.dart]: ?

File: lib/src/blocs/detail_screen_bloc/detail_bloc.dart [hunk 16]
```diff
@@ -0,0 +1,21 @@
...
+
+class DetailBloc extends Bloc<DetailEvent, DetailState> {
+  final DetailService? service;
```
Comment (thanhphamtrung) [inline lib/src/blocs/detail_screen_bloc/detail_bloc.dart:8]: can service be null?

File: lib/src/screens/detail_screen/detail_screen.dart [hunk 17]
```diff
@@ -4,11 +4,23 @@ import 'package:flutter/material.dart';
...
+import 'package:flutter_bloc/flutter_bloc.dart';
+
+import 'package:intl/intl.dart' as intl;
+
+import '/../src/blocs/detail_screen_bloc/detail_event.dart';
+import '/../src/blocs/detail_screen_bloc/detail_bloc.dart';
+import '/../src/blocs/detail_screen_bloc/detail_state.dart';
+
+import '/../src/widgets/custom_card_detail.dart';
+
```
Comment (thanhphamtrung) [inline lib/src/screens/detail_screen/detail_screen.dart]: ?

File: lib/src/services/detail_service/app_detail_service.dart [hunk 18]
```diff
@@ -0,0 +1,34 @@
...
+import 'dart:async';
+
+import 'package:http/http.dart' as http;
+
+import '/../src/config/app_config.dart';
+import '/../src/config/constants.dart';
+
+import '/../src/models/coins_detail.dart';
+
+import '/../src/services/detail_service/detail_service.dart';
```
Comment (thanhphamtrung) [inline lib/src/services/detail_service/app_detail_service.dart]: ?

**Review by thanhphamtrung — APPROVED** [review]
LGTM

**Pull request merged**